stdlib/public/RemoteInspection/TypeRefBuilder.cpp

```diff
@@ -471,6 +471,42 @@ TypeRefBuilder::ReflectionTypeDescriptorFinder::getBuiltinTypeInfo(
   return nullptr;
 }
 
+class BuiltinTypeDescriptorImpl : public BuiltinTypeDescriptorBase {
+  RemoteRef<BuiltinTypeDescriptor> BTD;
+  TypeRefBuilder &Builder;
+
+public:
+  BuiltinTypeDescriptorImpl(RemoteRef<BuiltinTypeDescriptor> BTD,
+                            TypeRefBuilder &Builder)
+      : BuiltinTypeDescriptorBase(BTD->Size, BTD->getAlignment(),
+                                       BTD->Stride, BTD->NumExtraInhabitants,
+                                       BTD->isBitwiseTakable()),
+        BTD(BTD), Builder(Builder) {}
+
+  ~BuiltinTypeDescriptorImpl() override {}
+
+  StringRef getMangledTypeName() override {
+    return Builder.getTypeRefString(Builder.readTypeRef(BTD, BTD->TypeName));
+  };
+};
+
+std::unique_ptr<BuiltinTypeDescriptorBase>
+TypeRefBuilder::ReflectionTypeDescriptorFinder::getBuiltinTypeDescriptor(
+    const TypeRef *TR) {
+  if (auto BTI = getBuiltinTypeInfo(TR))
+    return std::make_unique<BuiltinTypeDescriptorImpl>(BTI, Builder);
+  return nullptr;
+}
+
+std::unique_ptr<BuiltinTypeDescriptorBase>
+TypeRefBuilder::getBuiltinTypeDescriptor(const TypeRef *TR) {
+  if (DF) 
```

Comment:
Since you are going to write a bunch these, maybe you should have a SmallVector<DescriptorFInderBase&, 2> DFs and say 

```
for (DF : DFs) 
  if (auto D = DF->getBuiltinTypeDescriptor(TR))
      return D;
```
?

Reply:
Maybe it's overkill because you'll have exactly 2...